core/internal/testutils/testutils.go

```diff
@@ -371,8 +371,8 @@ func WaitForLogMessage(t *testing.T, observedLogs *observer.ObservedLogs, msg st
 // WaitForLogMessageCount waits until at least count log message containing the
 // specified msg is emitted
 func WaitForLogMessageCount(t *testing.T, observedLogs *observer.ObservedLogs, msg string, count int) {
-	i := 0
```

Comment:
This was accumulating across attempts, so it could yield false positives if at least one instance appeared.